core/src/main/java/hudson/views/JobColumn.java

```diff
@@ -36,6 +39,24 @@
     public JobColumn() {
     }
 
+    public String columnCaption(View view) {
+        String metaName = null;
+        for (Item item : view.getItems()) {
+            String m = item instanceof AbstractItem
+                    ? ((AbstractItem) item).getMetaName()
+                    : Messages.JobColumn_DisplayName();
+            if (metaName == null) {
+                metaName = m;
+            } else if (!metaName.equals(m)) {
+                return Messages.JobColumn_DisplayName();
+            }
+        }
+        if (metaName == null) {
```

Comment:
I would use StringUtils.isBlank(). Just to avoid blank column headers in the case of misbehaving <code>AlternativeUiTextProvider</code>s